board/src/pages/messages/conversation.js

```diff
@@ -35,31 +38,75 @@ export const NewConversationLine = () => (
   </Message>
 )
 
-const ConversationLine = ({ conversation, history }) => (
-  <Message
-    style={{ cursor: 'pointer' }}
-    onClick={e => {
-      history.push(`/messages/${conversation.id}`)
-    }}
-    active={history.location.pathname === `/messages/${conversation.id}`}
-  >
-    <SenderInfo>
-      <SenderCircle image={conversation.service.agency.logo}>
-        {conversation.service.agency.name.substring(0, 2)}
-      </SenderCircle>
-    </SenderInfo>
+const CounterSender = Counter(ShortSender)
 
-    <MessageContentWrapper>
-      <MessageContent>
-        <ShortSender>{conversation.service.agency.name}</ShortSender>
-        <ShortSubject>{conversation.subject}</ShortSubject>
-        <Timestamp dateTime={conversation.createdAt}>
-          {format(conversation.createdAt, 'ddd D MMM YYYY, h:mm a')}
-        </Timestamp>
-      </MessageContent>
-    </MessageContentWrapper>
-  </Message>
-)
+const markAsRead = gql`
+  mutation($conversationID: ID!, $now: String) {
+    updateManyMessages(
+      where: { conversation: { id: $conversationID } }
+      data: { readStatus: Read, readAt: $now }
+    ) {
+      count
+    }
+  }
+`
+
+const ConversationLine = ({ conversation, history }) => {
+  let c
+  const count = conversation.messages.filter(msg => msg.readStatus !== 'Read')
+    .length
+
+  return (
+    <Mutation
+      mutation={markAsRead}
+      onCompleted={e => {
+        c.resetStore()
```

Comment:
Is this the way they recommend to do this? i.e. make `c` a reference to client when you have it and then use it in `onCompleted` and hope it's there? Wouldn't `onCompleted` pass the client too?